ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +250,144 @@ void PowerSystemNode::injectFault (const string& fault_name,
     {
       ros::param::getCached("/faults/" + FAULT_NAME_HPD, hpd_wattage);
       wattage += hpd_wattage;
-    } else
+    }
+  }
+}
+
+void PowerSystemNode::injectCustomFault(bool& fault_activated,
+                                        const PrognoserVector& sequence,
+                                        size_t& index,
+                                        double& wattage,
+                                        double& voltage,
+                                        double& temperature)
+{
+  static string saved_fault_directory = "N/A";
+  string current_fault_directory;
+  string designated_file;
+  static string saved_file;
+  static bool custom_fault_ready = false;
+  static bool custom_warning_displayed = false;
+  static bool end_fault_warning_displayed = false;
+  bool fault_enabled = false;
+
+  // Do nothing unless the specified fault has been injected.
+  if (! ros::param::getCached("/faults/activate_custom_fault", fault_enabled)) {
+    return;
+  }
+
+  if (!fault_activated && fault_enabled)
+  {
+    // Multiple potential points of failure, so alert user the process has started.
+    ROS_INFO_STREAM("Attempting custom fault activation...");
+
+    // Get user-entered file directory.
+    if (! ros::param::getCached("/faults/custom_fault_profile", designated_file))
```

Comment:
Good catch, @AstroStucky.  @Lumsterling I wonder if this might explain some of the "flukey" behavior we've seen before.

Reply:
Interesting. This is another holdover from the original code; I just left it in without modifying it since it worked. I assumed it stored the parameter if found and returned true if so, and false if it couldn't find it. If it doesn't affect the behavior of the program at all I can remove it.